File: app/src/main/java/zebrostudio/wallr100/ui/explore/ExploreFragment.java
package zebrostudio.wallr100.ui.explore;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.View;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;
import zebrostudio.wallr100.R;
import zebrostudio.wallr100.di.ActivityScope;
import zebrostudio.wallr100.di.ApplicationContext;
import zebrostudio.wallr100.ui.main.MainActivity;
import zebrostudio.wallr100.utils.FragmentTags;
import zebrostudio.wallr100.utils.UiCustomizationHelper;

@ActivityScope
public class ExploreFragment extends DaggerFragment implements ExploreFragmentContract.ExploreView {

    @Inject
    MainActivity mMainActivity;
    @Inject
    ExplorePresenterImpl mExplorePresenter;
    @Inject
    UiCustomizationHelper mUiCustomizationHelper;
    @ApplicationContext
    @Inject
    Context mApplicationContext;

    @Inject
    public ExploreFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mExplorePresenter.bindView(this);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

    @Override
    public void onResume() {
        super.onResume();
        setUpUi();
        Log.d(FragmentTags.EXPLORE_FRAGMENT_TAG,"onresume");
        mExplorePresenter.updateCurrentFragmentTag();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mExplorePresenter.unbindView();
    }

    private void setUpUi() {
        mMainActivity.setTitle(FragmentTags.EXPLORE_FRAGMENT_TAG);
        mMainActivity.setTitlePadding(0, 0, 0, 0);
        mUiCustomizationHelper.showSearchOption();
        mUiCustomizationHelper.hideMultiSelectOption();
        mUiCustomizationHelper.hideSmartTabLayout();
        mUiCustomizationHelper.hideCollectionSwitchLayout();
        mUiCustomizationHelper.hideMinimalBottomPanel();
    }

}

File: app/src/main/java/zebrostudio/wallr100/ui/explore/ExploreFragment.kt
package zebrostudio.wallr100.ui.explore

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import zebrostudio.wallr100.R
import zebrostudio.wallr100.ui.basefragment.BaseFragment
import javax.inject.Inject

class ExploreFragment : BaseFragment(), ExploreContract.ExploreView {

  @Inject
  internal lateinit var presenter: ExplorePresenterImpl

  companion object {
    val EXPLORE_FRAGMENT_TAG = "Explore"

    fun newInstance(): ExploreFragment {
      return ExploreFragment()
    }
  }

  override fun onCreateView(
    inflater: LayoutInflater,
    container: ViewGroup?,
    savedInstanceState: Bundle?
  ): View? {
    return inflater.inflate(R.layout.fragment_layout, container, false)
  }

  override fun onResume() {
    super.onResume()
    presenter.attachView(this)
    presenter.updateFragmentName(EXPLORE_FRAGMENT_TAG)
  }

  override fun onDestroy() {
    presenter.detachView()
    super.onDestroy()
  }

}